MaxText/utils/ckpt_conversion/examples/convert_gemma2_to_mt.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+export HF_AUTH_TOKEN=""
+
+DATE=$(date +%Y-%m-%d)
+# config.base_output_directory
+OUTPUT_BASE_DIR=""
+# Model name must be consistent in utils/utils.py
+MODEL_NAME="gemma2-2b"
+# Tokenizer path for decoding
+TOKENIZER_PATH="assets/tokenizer.gemma"
+
+PER_DEVICE_BATCH_SIZE=1
+ASYNC_CHECKPOINTING=false
+SCAN_LAYERS=false
+PROMPT="I love to"
+
+# --- Step 1: Convert Checkpoint to MaxText Format ---
+echo "--- Starting Checkpoint Conversion ---"
+python3 -m "MaxText.utils.ckpt_conversion.to_maxtext" \
+  "MaxText/configs/base.yml" \
+  model_name="${MODEL_NAME}" \
+  base_output_directory="${OUTPUT_BASE_DIR}" \
+  per_device_batch_size="${PER_DEVICE_BATCH_SIZE}" \
+  run_name="mt_gemma2" \
+  async_checkpointing="${ASYNC_CHECKPOINTING}" \
+  scan_layers="${SCAN_LAYERS}" \
+  prompt="${PROMPT}" \
+  attention='dot_product'
+
+echo "--- Checkpoint Conversion Complete ---"
+
+# --- Step 2 (Optional): Decode using the Converted Checkpoint ---
+
+echo "--- Starting Decoding ---"
+python3 -m "MaxText.decode" \
+  "MaxText/configs/base.yml" \
+  model_name="${MODEL_NAME}" \
+  tokenizer_path="${TOKENIZER_PATH}" \
+  load_parameters_path="${OUTPUT_BASE_DIR}/0/items" \
+  per_device_batch_size="${PER_DEVICE_BATCH_SIZE}" \
+  run_name="mt_gemma2_check" \
+  max_prefill_predict_length=8 \
+  max_target_length=16 \
+  dataset_type=synthetic \
+  steps=1 \
+  async_checkpointing="${ASYNC_CHECKPOINTING}" \
+  scan_layers="${SCAN_LAYERS}" \
+  prompt="${PROMPT}"
+
+echo "--- Decoding Complete ---"
+
+# --- Step 3: Compare the HF and MT Checkpoint ---
+
+echo "--- Starting Comparing Logits and Predicted Tokens ---"
+python3 -m "MaxText.tests.mt_hf_mutual_conversion_check" \
+    hf_model_id="google/gemma-2-2b" \
```

Comment:
could you make this a parameter, also below instead of hardcoding, so all model-specific parameters are at the top of this script